File: Python Scripts/ai_insight.py
import requests
import pandas as pd

# Paste your OpenRouter API key here
API_KEY = "YOUR_OPENROUTER_API_KEY"

# Load cleaned dataset
df = pd.read_csv("cleaned_superstore.csv")

# Create real dataset summary
total_sales = round(df["Sales"].sum(), 2)
average_sales = round(df["Sales"].mean(), 2)
top_category = df.groupby("Category")["Sales"].sum().idxmax()
top_region = df.groupby("Region")["Sales"].sum().idxmax()

# AI prompt
prompt = f"""
You are a business data analyst.

Analyze this retail dataset:

Total Sales: {total_sales}
Average Sales: {average_sales}
Top Category: {top_category}
Top Region: {top_region}

Write a short business insight and one recommendation.
"""

# Send request to OpenRouter AI
response = requests.post(
    url="https://openrouter.ai/api/v1/chat/completions",
    headers={
        "Authorization": f"Bearer {API_KEY}",
        "Content-Type": "application/json"
    },
    json={
        "model": "openrouter/free",
        "messages": [
            {
                "role": "user",
                "content": prompt
            }
        ]
    }
)

# Convert response to JSON
result = response.json()

print("\nFULL API RESPONSE:\n")
print(result)

# Check if AI response exists
if "choices" in result:
    print("\nAI GENERATED INSIGHT:\n")
    print(result["choices"][0]["message"]["content"])
else:
    print("\nERROR FROM API:")
    print(result)

File: api/ai-insight.js
export default async function handler(req, res) {
  const prompt = `
You are a business data analyst.

Analyze this Superstore dashboard summary:
Total Sales: 805,853.84
Average Sales: 91.58
Top Category: Office Supplies
Regions: West, East, Central, South

Write a short business insight and one recommendation.
`;

  try {
    const response = await fetch("https://openrouter.ai/api/v1/chat/completions", {
      method: "POST",
      headers: {
        "Authorization": `Bearer ${process.env.OPENROUTER_API_KEY}`,
        "Content-Type": "application/json"
      },
      body: JSON.stringify({
        model: "openrouter/free",
        messages: [
          {
            role: "user",
            content: prompt
          }
        ]
      })
    });

    const data = await response.json();

    res.status(200).json({
      insight: data.choices?.[0]?.message?.content || "No AI insight generated."
    });

  } catch (error) {
    res.status(500).json({
      insight: "Error generating AI insight."
    });
  }
}